en/Sorting Algorithms/Binary Insertion Sort.md

```diff
@@ -0,0 +1,129 @@
+
+#### Example 
+
+Consider the array 29, 10, 14, 37, 14,
+
+![binary insertion sort - array](https://cdn.hashnode.com/res/hashnode/image/upload/v1661929956266/DSLKdODLR.png align="center")
+
+
+**First Pass**
+
+Key = 1
+
+Since we consider the first element is in the sorted array, we will be starting from the second element. Then we apply the binary search on the sorted array. 
+
+![first pass - binary insertion sort ](https://cdn.hashnode.com/res/hashnode/image/upload/v1661934538052/qqF1rB3Qq.png align="center")
+
+In this scenario, we can see that the middle element in sorted array  (29) is greater than the key element 10. So the position of the key element is 0. Then we can shift the remaining elements by 1 position.
+
+
+![after binary search first pass - binary insertion sort](https://cdn.hashnode.com/res/hashnode/image/upload/v1661934765696/rVvhl09gD.png align="center")
+
+Increment the value of key. 
+
+**Second Pass**
+
+Key = 2
+
+Now the key element is 14. We will apply binary search in the sorted array to find the position of the key element. 
+
+
+![binary insertion sort -  second pass](https://cdn.hashnode.com/res/hashnode/image/upload/v1661934886050/hO3y2uCQ7.png align="center")
+
+In this scenario, by applying binary search, we can see key element to be placed at index 1 (between 10 and 29). Then we can shift the remaining elements by 1 position.
+
+
+![binary insertion sort - after binary search, second pass](https://cdn.hashnode.com/res/hashnode/image/upload/v1661934956931/sbe6VJ_2K.png align="center")
+
+
+**Third Pass**
+
+Key = 3
+
+Now the key element is 37. We will apply binary search in the sorted array to find the position of the key element. 
+
+
+![binary insertion sort - third pass](https://cdn.hashnode.com/res/hashnode/image/upload/v1661935075142/N6zvcH73Q.png align="center")
+
+In this scenario, by applying binary search, we can see key element is placed in its correct position. 
+
+![binary insertion sort - after binary search](https://cdn.hashnode.com/res/hashnode/image/upload/v1661935158835/52ZGwrAOF.png align="center")
+
+
+**Fourth Pass**
+
+Key = 4
+
+Now the key element is 14. We will apply binary search in the sorted array to find the position of the key element. 
+
+
+![binary insertion sort - fourth pass ](https://cdn.hashnode.com/res/hashnode/image/upload/v1661935242311/YX7lhAazo.png align="center")
+
+In this scenario, by applying binary search, we can see key element to be placed at index 2 (between 14 and 29). Then we can shift the remaining elements by 1 position.
+
+
+![binary insertion sort - after binary search](https://cdn.hashnode.com/res/hashnode/image/upload/v1661935283231/zCdCYPErq.png align="center")
+
+Now we can see all the elements are sorted.
+
+#### Is Binary Insertion Sort a stable algorithm ?
+It is a stable sorting algorithm, the elements with the same values appear in the same order in the final array as they were in the initial array.
```

Comment:
I doubt this. For this property to be provided, you'd need the binary search to guarantee that it always returns the rightmost insertion index. This requirement must at the very least be made explicit. Ideally it should also be noted how this can be implemented.